mongodb/vibe/db/mongo/connection.d

```diff
@@ -183,7 +186,9 @@ final class MongoConnection {
 			throw new MongoDriverException(format("Failed to connect to MongoDB server at %s:%s.", m_settings.hosts[0].name, m_settings.hosts[0].port), __FILE__, __LINE__, e);
 		}
 
-		if (m_settings.nextGen) {
+		m_allowReconnect = false;
+		scope (exit)
+			m_allowReconnect = true;
```

Comment:
failed a bit partially committing my changes here, but it's fixed in the later commits